File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

from typing import List

# @lc code=start


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []

        for idx, x in enumerate(nums):
            firstNumber = x
            secondNumber = target - firstNumber

            for idy, y in enumerate(nums[idx+1:]):
                anotherNumber = y

                if (anotherNumber == secondNumber):
                    result.insert(0, idx)
                    result.insert(1, idy+idx+1)
                    return result

# @lc code=end

if __name__ == '__main__':
    solution = Solution()
    print(solution.twoSum([2,7,11,15], 9))

File: [1] Two Sum/1.two-sum.java
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];

        for (int k = 0; k < nums.length; k++) {
            int firstNumber = nums[k];
            int secondNumber = target - firstNumber;

            for (int l = k + 1; l < nums.length; l++) {
                int anotherNumber = nums[l];

                if (anotherNumber == secondNumber) {
                    result[0] = k;
                    result[1] = l;
                    break;
                }
            }
        }

        return result;
    }
}
// @lc code=end
